v2/inc/c_logging/logger.h

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef LOGGER_H
+#define LOGGER_H
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/log_context.h"
+
+#define LOG_LEVEL_VALUES \
+    LOG_LEVEL_CRITICAL, \
+    LOG_LEVEL_ERROR, \
+    LOG_LEVEL_WARNING, \
+    LOG_LEVEL_INFO, \
+    LOG_LEVEL_VERBOSE
+
+MU_DEFINE_ENUM_WITHOUT_INVALID(LOG_LEVEL, LOG_LEVEL_VALUES);
+
+#define LOG_MAX_MESSAGE_LENGTH 4096 /*in bytes - a message is not expected to exceed this size in bytes, if it does, only LOG_MAX_MESSAGE_LENGTH characters are retained*/
+
+void logger_log(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, const char* format, ...);
```

Comment:
>logger_log [](http://example.com/codeflow?start=5&length=10)

as with any lib that has a elipsys API, it should have another one that does the same thing taking a va_list.

Reply:
A very good ideea, shall add, will leave the comment open until it's done.

Reply:
I will actually hold off on the va_args flavor for now.
That can always be added later (just so I get this moving).